include/class.thread.php

```diff
@@ -1539,25 +1555,38 @@ function lookupByRefMessageId($mid, $from) {
     }
 
     function setExtra($entries, $info=NULL, $thread_id=NULL) {
-        foreach ($entries as $entry)
+        foreach ($entries as $entry) {
+            $mergeInfo = new ThreadEntryMergeInfo(array(
+                'thread_entry_id' => $entry->getId(),
+                'data' => json_encode($info),
+            ));
+            $mergeInfo->save();
             $entry->saveExtra($info, $thread_id);
+        }
+
     }
 
     function saveExtra($info=NULL, $thread_id=NULL) {
-        if (!$this->extra) {
-            $this->extra = !is_null($info) ? json_encode($info) : NULL;
-            $this->setFlag(ThreadEntry::FLAG_CHILD, true);
-        }
-        $this->thread_id = $thread_id ?: $thread_id;
+        $this->setFlag(ThreadEntry::FLAG_CHILD, true);
+        $this->thread_id = $thread_id;
         $this->save();
     }
 
+    function getMergeData() {
+        $row = ThreadEntryMergeInfo::objects()
```

Comment:
Why are you doing direct lookup here? thread entry already has a relationship.

```php
return $this->merge_info ? $this->merge_info->data : null;
```

Reply:
That makes more sense... My bad lol